### ISSUE TITLE
Update/replace System.Threading.Tasks.Dataflow internal ConcurrentQueue with Concurrent.Collections changes

### ISSUE THREAD

**Issue by nietras** [dotnet/corefx#17964]
Apparently, DataFlow has it's own copy of `ConcurrentQueue`, but this does not seem to have been updated with the great optimizations that @stephentoub did to the Concurrent.Collections version see:

https://github.com/dotnet/corefx/blob/master/src/System.Threading.Tasks.Dataflow/src/Internal/ConcurrentQueue.cs

Either update this with the changes or use Concurrent.Collections.

**Comment by stephentoub** [comment]
That's private copy is only used when building for a platform that doesn't have ConcurrentQueue, namely an old portable profile.  I'm not even sure if we still need to maintain that build. @terrajobst, @weshaggard?

**Comment by svick** [comment]
@stephentoub [According to the project file, it's used in the .Net Standard 1.0 version](https://github.com/dotnet/corefx/blob/master/src/System.Threading.Tasks.Dataflow/src/System.Threading.Tasks.Dataflow.csproj#L58-L59) ([which indeed does not have `ConcurrentQueue`](https://apisof.net/catalog/System.Collections.Concurrent.ConcurrentQueue%3CT%3E)).

**Comment by stephentoub** [comment]
Right, that's what I meant by old portable profile.  I'm not sure if/why we still need to be building this for that.

**Comment by alexperovich** [comment]
AFAIK if we stop building the netstandard1.0 version then we will just pick up the asset from the latest shipped package and include that. I think that would be fine.

**Comment by weshaggard** [comment]
>AFAIK if we stop building the netstandard1.0 version then we will just pick up the asset from the latest shipped package and include that. I think that would be fine.

Yes we will just pick it up from the last stable package. 

As for the general case of the older configurations it is left up to the owner of the component whether or not they want to continue building and shipping fixes for it in our latest release, or if they want to consider it done and not touch it except for possibly in a servicing release. Each library has slightly different requirements and constraints which is why the owner is the one that should make the call.
